### PR TITLE
Refactor out SpoonFilter::getGetValue and SpoonFilter::getPostValue

### LINKED ISSUE #1347: refactor out SpoonFilter::getGetValue and SpoonFilter::getPostValue
These methods access the $_GET and $_POST superglobals, which make writing tests a lot harder.

We can get our GET and POST variables from the request object which can be fetched everywhere.

### PR DESCRIPTION
**Pull request by jonasdekeukelaere**
- Add shortcut to get request from container
- Refactor out SpoonFilter::getGetValue
- Refactor out SpoonFilter::getPostValue
- Refactor out $_GET
- Refactor out $_POST

Some of the $_GET and $_POST are still in the codebase, but that has to do with the fact that SpoonForm still uses $_GET and $_POST directly, so to manipulate the values we need to directly alter the superglobals.
Fixes #1347

### CODE REVIEW

File: src/Backend/Modules/FormBuilder/Actions/MassDataAction.php [hunk 1]
```diff
@@ -39,7 +42,7 @@ public function execute(): void
             $this->redirect(BackendModel::createURLForAction('Index') . '&error=non-existing');
```
Comment (carakas) [inline src/Backend/Modules/FormBuilder/Actions/MassDataAction.php]: I would also do this at the other actions where you had the only actions as default

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php [hunk 2]
```diff
@@ -34,12 +34,24 @@ public function execute(): void
...
         // get parameters
-        $language = \SpoonFilter::getPostValue('language', array_keys($possibleLanguages), null, 'string');
-        $module = \SpoonFilter::getPostValue('module', BackendModel::getModules(), null, 'string');
-        $name = \SpoonFilter::getPostValue('name', null, null, 'string');
-        $type = \SpoonFilter::getPostValue('type', BackendModel::getContainer()->get('database')->getEnumValues('locale', 'type'), null, 'string');
-        $application = \SpoonFilter::getPostValue('application', ['Backend', 'Frontend'], null, 'string');
-        $value = \SpoonFilter::getPostValue('value', null, null, 'string');
+        $language = $this->getRequest()->request->get('language');
+        if (!array_key_exists($language, $possibleLanguages)) {
+            $language = null;
```
Comment (carakas) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: the default should be '' not null unless you change the check below

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php [hunk 3]
```diff
@@ -34,12 +34,24 @@ public function execute(): void
...
+        }
+        $module = $this->getRequest()->request->get('module');
+        if (!in_array($module, BackendModel::getModules())) {
+            $module = null;
```
Comment (carakas) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: the default should be '' not null unless you change the check below

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php [hunk 4]
```diff
@@ -34,12 +34,24 @@ public function execute(): void
...
+        }
+        $name = $this->getRequest()->request->get('name', '');
+        $type = $this->getRequest()->request->get('type');
+        if (!in_array($type, BackendModel::get('database')->getEnumValues('locale', 'type'))) {
+            $type = null;
```
Comment (carakas) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: the default should be '' not null unless you change the check below

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php [hunk 5]
```diff
@@ -34,12 +34,24 @@ public function execute(): void
...
+        }
+        $application = $this->getRequest()->request->get('application');
+        if (!in_array($application, ['Backend', 'Frontend'])) {
+            $application = null;
```
Comment (carakas) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: the default should be '' not null unless you change the check below

File: src/Backend/Modules/Location/Ajax/SaveLiveLocation.php [hunk 6]
```diff
@@ -25,30 +25,35 @@ public function execute(): void
...
-        $mapStyle = trim(\SpoonFilter::getPostValue('style', ['standard', 'custom', 'gray', 'blue'], 'standard'));
-        $centerLat = \SpoonFilter::getPostValue('centerLat', null, 1, 'float');
-        $centerlng = \SpoonFilter::getPostValue('centerLng', null, 1, 'float');
-        $height = \SpoonFilter::getPostValue('height', null, $generalSettings['height'], 'int');
-        $width = \SpoonFilter::getPostValue('width', null, $generalSettings['width'], 'int');
-        $showLink = \SpoonFilter::getPostValue('link', ['true', 'false'], 'false', 'string');
-        $showDirections = \SpoonFilter::getPostValue('directions', ['true', 'false'], 'false', 'string');
-        $showOverview = \SpoonFilter::getPostValue('showOverview', ['true', 'false'], 'true', 'string');
+        $itemId = $this->getRequest()->request->getInt('id');
+        $zoomLevel = $this->getRequest()->request->get('zoom', 'auto');
```
Comment (carakas) [inline src/Backend/Modules/Location/Ajax/SaveLiveLocation.php]: the `trim()` is missing here

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 7]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
                 null,
                 null,
                 [
-                     'offset' => \SpoonFilter::getGetValue('offset', null, ''),
-                     'order' => \SpoonFilter::getGetValue('order', null, ''),
-                     'sort' => \SpoonFilter::getGetValue('sort', null, ''),
-                     'email' => \SpoonFilter::getGetValue('email', null, ''),
-                     'status' => \SpoonFilter::getGetValue('status', null, ''),
-                     'group' => \SpoonFilter::getGetValue('group', null, ''),
+                     'offset' => $this->getRequest()->get('offset', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 8]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
...
+                     'order' => $this->getRequest()->get('order', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 9]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
...
+                     'sort' => $this->getRequest()->get('sort', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 10]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
...
+                     'email' => $this->getRequest()->get('email', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 11]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
...
+                     'status' => $this->getRequest()->get('status', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Backend/Modules/Profiles/Actions/MassAction.php [hunk 12]
```diff
@@ -86,12 +92,12 @@ public function execute(): void
...
+                     'group' => $this->getRequest()->get('group', ''),
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: ->query

File: src/Frontend/Modules/Profiles/Engine/Model.php [hunk 13]
```diff
@@ -336,7 +336,9 @@ public static function parse(): void
         ];
 
         // query string
-        $queryString = (isset($_GET['queryString'])) ? SITE_URL . '/' . urldecode($_GET['queryString']) : SELF;
+        $queryString = self::get('Request')->query->has('queryString')
+            ? SITE_URL . '/' . urldecode(self::get('Request')->query->get('queryString'))
+            : SELF;
```
Comment (carakas) [inline src/Frontend/Modules/Profiles/Engine/Model.php]: is it possible that this needs to be `self` ?

File: src/Frontend/Modules/Search/Actions/Index.php [hunk 14]
```diff
@@ -210,8 +210,8 @@ private function buildForm(): void
         $this->form = new FrontendForm('search', null, 'get', null, false);
 
         // could also have been submitted by our widget
-        if (!\SpoonFilter::getGetValue('q', null, '')) {
-            $_GET['q'] = \SpoonFilter::getGetValue('q_widget', null, '');
+        if (!$this->getRequest()->has('q')) {
+            $_GET['q'] = $this->getRequest()->query->get('q_widget');
```
Comment (carakas) [inline src/Frontend/Modules/Search/Actions/Index.php]: I think you will have to look at the logic here because I don't think the request object keeps checking the `$_GET`

File: docs/05. module guide/13. ajax.md [hunk 15]
```diff
@@ -35,7 +35,7 @@ class ThisIsAwesome extends FrontendBaseAJAXAction
 {
     public function execute()
     {
-        $post_id = (int)\SpoonFilter::getPostValue('post_id', null, '', 'int');
+        $post_id = $this->getRequest()->request->getInt('post_id');
 
         if ($post_id == 0) {
```
Comment (jonasdekeukelaere) [inline docs/05. module guide/13. ajax.md]: this MR is not about that :/

File: docs/05. module guide/23. api.md [hunk 16]
```diff
@@ -15,14 +15,15 @@ In our case we'll just have one method, `showTopAwesome`. This method will retur
 namespace Backend\Modules\MiniBlog\Engine;
 
 use Api\V1\Engine\Api as BaseAPI;
```
Comment (jonasdekeukelaere) [inline docs/05. module guide/23. api.md]: this MR is not about that :/

File: src/Backend/Modules/Blog/Actions/Add.php [hunk 17]
```diff
@@ -64,7 +64,7 @@ private function loadForm(): void
         $this->frm->addEditor('introduction');
         $this->frm->addRadiobutton('hidden', $rbtHiddenValues, 'N');
         $this->frm->addCheckbox('allow_comments', $this->get('fork.settings')->get($this->getModule(), 'allow_comments', false));
-        $this->frm->addDropdown('category_id', $categories, \SpoonFilter::getGetValue('category', null, null, 'int'));
+        $this->frm->addDropdown('category_id', $categories, $this->getRequest()->query->getInt('category'));
         if (count($categories) != 2) {
```
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Add.php]: this MR is not about that :/

File: src/Backend/Modules/Blog/Actions/Delete.php [hunk 18]
```diff
@@ -37,7 +37,7 @@ public function execute(): void
             parent::execute();
 
             // set category id
-            $this->categoryId = \SpoonFilter::getGetValue('category', null, null, 'int');
+            $this->categoryId = $this->getRequest()->query->getInt('category');
             if ($this->categoryId == 0) {
```
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Delete.php]: this MR is not about that :/

File: src/Backend/Modules/Blog/Actions/Edit.php [hunk 19]
```diff
@@ -60,7 +60,7 @@ public function execute(): void
             parent::execute();
 
             // set category id
-            $this->categoryId = \SpoonFilter::getGetValue('category', null, null, 'int');
+            $this->categoryId = $this->getRequest()->query->getInt('category');
             if ($this->categoryId == 0) {
```
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Edit.php]: this MR is not about that :/

File: src/Backend/Modules/Extensions/Actions/Themes.php [hunk 20]
```diff
@@ -79,7 +79,9 @@ private function loadForm(): void
         $themes = $this->installedThemes;
 
         // set selected theme
-        $selected = isset($_POST['installedThemes']) ? $_POST['installedThemes'] : $this->get('fork.settings')->get('Core', 'theme', 'core');
+        $selected = $this->getRequest()->request->has('installedThemes')
+            ? $this->getRequest()->request->get('installedThemes')
+            : $this->get('fork.settings')->get('Core', 'theme', 'core');
```
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Extensions/Actions/Themes.php]: Just copied it from existing code

File: src/Backend/Modules/Blog/Actions/Index.php [hunk 21]
```diff
@@ -51,7 +51,7 @@ public function execute(): void
         parent::execute();
 
         // set category id
-        $this->categoryId = \SpoonFilter::getGetValue('category', null, null, 'int');
+        $this->categoryId = $this->getRequest()->query->getInt('category');
         if ($this->categoryId == 0) {
```
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Index.php]: this MR is not about that :/

File: src/Backend/Modules/FormBuilder/Actions/MassDataAction.php (hunk 1 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/FormBuilder/Actions/MassDataAction.php]: this MR is not about that :/

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 2 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: it was null by default

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 3 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: it was null by default

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 4 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: it was null by default

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 5 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: it was null by default

File: src/Frontend/Modules/Profiles/Engine/Model.php (hunk 13 above)
Comment (jonasdekeukelaere) [inline src/Frontend/Modules/Profiles/Engine/Model.php]: just copied from existing code

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 2 above)
Comment (carakas) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: @jonasdekeukelaere can you please read the code and what it does instead of being stubborn when I point out that you are introducing a bug

File: src/Backend/Modules/FormBuilder/Actions/MassDataAction.php (hunk 1 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/FormBuilder/Actions/MassDataAction.php]: Should be fixed in https://github.com/forkcms/forkcms/issues/2050

File: docs/05. module guide/13. ajax.md (hunk 15 above)
Comment (jonasdekeukelaere) [inline docs/05. module guide/13. ajax.md]: Fixed cfc6e8e

File: src/Backend/Modules/Blog/Actions/Add.php (hunk 17 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Add.php]: Fixed cfc6e8e

File: src/Backend/Modules/Blog/Actions/Delete.php (hunk 18 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Delete.php]: Fixed cfc6e8e

File: src/Backend/Modules/Blog/Actions/Edit.php (hunk 19 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Edit.php]: Fixed cfc6e8e

File: src/Backend/Modules/Blog/Actions/Index.php (hunk 21 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Blog/Actions/Index.php]: Fixed cfc6e8e

File: docs/05. module guide/23. api.md [hunk 22]
```diff
@@ -15,14 +15,15 @@ In our case we'll just have one method, `showTopAwesome`. This method will retur
...
+use Backend\Core\Engine\Model;
 use Backend\Modules\MiniBlog\Engine\Model as BackendMiniBlogModel;
 
 class Api
 {
     public static function showTopAwesome()
     {
-        $number = (int)\SpoonFilter::getGetValue('number', null, 10);
-        $days = (int)\SpoonFilter::getGetValue('days', null, 7);
+        $number = BackendModel::get('request')->query->getInt('number', 10);
```
Comment (jonasdekeukelaere) [inline docs/05. module guide/23. api.md]: API doc removed 7a2c2cb

File: docs/05. module guide/23. api.md (hunk 16 above)
Comment (jonasdekeukelaere) [inline docs/05. module guide/23. api.md]: Fixed 7a2c2cb

File: src/Backend/Modules/Location/Ajax/SaveLiveLocation.php (hunk 6 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Location/Ajax/SaveLiveLocation.php]: Fixed 4b9479c

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 7 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 8 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 9 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 10 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 11 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Modules/Profiles/Actions/MassAction.php (hunk 12 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Profiles/Actions/MassAction.php]: Fixed 583698f

File: src/Backend/Core/Language/Language.php [hunk 23]
```diff
@@ -201,10 +201,12 @@ public static function getLabels(): array
     public static function getMessage(string $key, string $module = null): string
     {
         if ($module === null) {
+            $request = Model::get('request');
```
Comment (jonasdekeukelaere) [inline src/Backend/Core/Language/Language.php]: Fixed 3f7d7dc

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 2 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: Fixed f5ce21f

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 3 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: Fixed f5ce21f

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 4 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: Fixed f5ce21f

File: src/Backend/Modules/Locale/Ajax/SaveTranslation.php (hunk 5 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Locale/Ajax/SaveTranslation.php]: Fixed f5ce21f

File: src/Backend/Modules/Extensions/Actions/Themes.php (hunk 20 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Extensions/Actions/Themes.php]: Set default theme to Fork ccf9c74

File: src/Frontend/Modules/Profiles/Engine/Model.php (hunk 13 above)
Comment (jonasdekeukelaere) [inline src/Frontend/Modules/Profiles/Engine/Model.php]: SELF has been refactor a long time ago, so fixed in 56a464e

File: src/Frontend/Modules/Search/Actions/Index.php (hunk 14 above)
Comment (jonasdekeukelaere) [inline src/Frontend/Modules/Search/Actions/Index.php]: Not sure if I understand what you mean. $_GET doesn't get altered before this line.

**Comment by jonasdekeukelaere** [comment]
@carakas all feedback has been handled

File: src/Backend/Core/Ajax/GenerateUrl.php [hunk 24]
```diff
@@ -22,10 +22,10 @@ public function execute(): void
         parent::execute();
 
         // get parameters
-        $url = \SpoonFilter::getPostValue('url', null, '', 'string');
-        $className = \SpoonFilter::getPostValue('className', null, '', 'string');
-        $methodName = \SpoonFilter::getPostValue('methodName', null, '', 'string');
-        $parameters = \SpoonFilter::getPostValue('parameters', null, '', 'string');
+        $url = $this->getRequest()->request->get('url', '');
```
Comment (jeroendesloovere) [inline src/Backend/Core/Ajax/GenerateUrl.php]: Would be better if the code uses the default `null`.
`$this->getRequest()->request->get('url');`
=> Maybe change further code in this method?

File: src/Backend/Modules/Authentication/Actions/ResetPassword.php [hunk 25]
```diff
@@ -61,8 +61,8 @@ public function execute(): void
     private function isThePasswordResetKeyCorrect(): bool
     {
         // catch the key and e-mail address from GET
-        $this->email = urldecode(\SpoonFilter::getGetValue('email', null, ''));
-        $this->key = \SpoonFilter::getGetValue('key', null, '');
+        $this->email = urldecode($this->getRequest()->query->get('email', ''));
+        $this->key = $this->getRequest()->query->get('key', '');
 
         // if the email or the key aren't set, redirect the user
         if ($this->email !== '' && $this->key !== '') {
```
Comment (jeroendesloovere) [inline src/Backend/Modules/Authentication/Actions/ResetPassword.php]: I prefer using null instead of `''`

File: src/Backend/Modules/Blog/Ajax/AddCategory.php [hunk 26]
```diff
@@ -23,7 +23,7 @@ public function execute(): void
         parent::execute();
 
         // get parameters
-        $categoryTitle = trim(\SpoonFilter::getPostValue('value', null, '', 'string'));
+        $categoryTitle = trim($this->getRequest()->request->get('value', ''));
```
Comment (jeroendesloovere) [inline src/Backend/Modules/Blog/Ajax/AddCategory.php]: I prefer using the default `null` instead of an empty string...

File: src/Backend/Core/Ajax/GenerateUrl.php (hunk 24 above)
Comment (jonasdekeukelaere) [inline src/Backend/Core/Ajax/GenerateUrl.php]: Don't think much can be changed here, as `$this->get('fork.repository.meta')->generateURL` expects strings

File: src/Backend/Modules/Authentication/Actions/ResetPassword.php (hunk 25 above)
Comment (jonasdekeukelaere) [inline src/Backend/Modules/Authentication/Actions/ResetPassword.php]: The problem with changing the ifs to `$this->email === null` is that when the value is actually empty (`''`) your check will fail, no?

Comment (carakas) [inline src/Backend/Modules/Authentication/Actions/ResetPassword.php]: you can change the !== to !empty($this->email) maybe?

but maybe you can clean this code up a bit better, the query has also the option to check if data is filled, maybe test what the has does with empty strings, cause in that case you would be able to run this check on the query of the request

File: src/Backend/Core/Ajax/GenerateUrl.php (hunk 24 above)
Comment (carakas) [inline src/Backend/Core/Ajax/GenerateUrl.php]: if it expects strings then we should pass strings indeed

File: src/Backend/Modules/Authentication/Actions/ResetPassword.php (hunk 25 above)
Comment (carakas) [inline src/Backend/Modules/Authentication/Actions/ResetPassword.php]: if that doesn't work I would leave it at this

Comment (jonasdekeukelaere) [inline src/Backend/Modules/Authentication/Actions/ResetPassword.php]: isn't empty($x) as dirty as setting the default to ''?
as for using the has-method, `$request->query->has('x')` gives true if you give `?x=`. so no success there.

File: docs/README.md [hunk 27]
```diff
@@ -76,7 +76,6 @@ Questions? Feel free to discuss on our [Slack channel](https://fork-cms.herokuap
 1. [Tags](docs/05. module guide/20. tags.md)
 1. [Search](docs/05. module guide/21. search.md)
 1. [Creating frontend-widgets](docs/05. module guide/22. creating_frontend_widgets.md)
-1. [API](docs/05. module guide/23. api.md)
 1. [How does the API work](docs/05. module guide/24. how_does_the_api_work.md)
```
Comment (carakas) [inline docs/README.md]: could you maybe also remove this one? didn't know there where 2

File: src/Backend/Modules/Extensions/Actions/Themes.php [hunk 28]
```diff
@@ -81,7 +81,7 @@ private function loadForm(): void
         // set selected theme
         $selected = $this->getRequest()->request->has('installedThemes')
             ? $this->getRequest()->request->get('installedThemes')
-            : $this->get('fork.settings')->get('Core', 'theme', 'core');
+            : $this->get('fork.settings')->get('Core', 'theme', 'Fork');
```
Comment (carakas) [inline src/Backend/Modules/Extensions/Actions/Themes.php]: is there already an issue to check if all the places uses the correct default?

Comment (jonasdekeukelaere) [inline src/Backend/Modules/Extensions/Actions/Themes.php]: Only issue I added was to remove "no-theme".

File: src/Frontend/Modules/Search/Actions/Index.php (hunk 14 above)
Comment (carakas) [inline src/Frontend/Modules/Search/Actions/Index.php]: SpoonFIlter ran directly on `$_GET` etc, but I don't think the request object works like that
Also below `$_GET` is used, can you change that as well? since it has to do with the request parameters

File: src/Backend/Modules/Blog/Ajax/AddCategory.php (hunk 26 above)
Comment (carakas) [inline src/Backend/Modules/Blog/Ajax/AddCategory.php]: since we are talking about a string an empty string makes more sense I think

File: docs/README.md (hunk 27 above)
Comment (jonasdekeukelaere) [inline docs/README.md]: Fixed 793034a614ac827cd9bf551021120a04132cb5aa

File: src/Frontend/Modules/Search/Actions/Index.php (hunk 14 above)
Comment (jonasdekeukelaere) [inline src/Frontend/Modules/Search/Actions/Index.php]: Fixed 39680c3

**Comment by jonasdekeukelaere** [comment]
@carakas all feedback has been handled
@jeroendesloovere as discussed above, we are not changing the default value

**Comment by jonasdekeukelaere** [comment]
@carakas I also removed the ->getParameter() method

**Comment by carakas** [comment]
@jonasdekeukelaere Code looks good but I'm still missing an entry in the upgrade guide

**Comment by jonasdekeukelaere** [comment]
@carakas Fixed in 70631a50d
